MediaBrowser.Providers/MediaInfo/AudioFileProber.cs

```diff
@@ -400,6 +400,30 @@ private async Task FetchDataFromTags(Audio audio, Model.MediaInfo.MediaInfo medi
                 }
             }
 
+            if (options.ReplaceAllMetadata || !audio.TryGetProviderId(MetadataProvider.MusicBrainzRecording, out _))
+            {
+                if (!track.AdditionalFields.TryGetValue("MUSICBRAINZ_TRACKID", out var musicBrainzRecordingIdTag)
+                     && !track.AdditionalFields.TryGetValue("MusicBrainz Recording Id", out musicBrainzRecordingIdTag)
+                     && track.AdditionalFields.TryGetValue("UFID", out musicBrainzRecordingIdTag))
+                {
+                    string[] parts = musicBrainzRecordingIdTag.Split("\0");
+                    if (parts[0] != "http://musicbrainz.org")
+                    {
+                        musicBrainzRecordingIdTag = null;
```

Comment:
I don't think this is entirely correct? It looks like UFID split by `\0` with the first part being musicbrainz url is only the ID3 specific value and on Vorbis tags and MP4 tags they are stored as-is without the url part, if I understand its mapping correctly. Can you check that if you have such files? All common FLAC files should be using Vorbis tags.

Reply:
My understanding is `UFID` is id3v2 only, vorbis would be using `MUSICBRAINZ_TRACKID`. I don't know if ATL let you know if the tag is coming from id3v2?

Reply:
> My understanding is that UFID is id3v2 only

That is exactly the problem. Only for id3v2 UFID tag you have to perform the split, and for other tagging systems the tagging value is written as is. Your current logic will simply put everything not tagged like UFID to null, isn't it?

> I don't know if ATL let you know if the tag is coming from id3v2?

It does but we should not care in this case. We just check the UFID case separately from the Voribs/MP4 tag so that they don't overwrite their values each other.

To simplify code we first check UFID, pre-process it by doing the splitting and get the second part, and then use the Vorbis/MP4 logic, if that UFID extracted value is empty.

Reply:
> not tagged like UFID to null, isn't it?

I don't think it would? `MUSICBRAINZ_TRACKID` should be handled fine?